examples/ddpg/ddpg.py

```diff
@@ -2,196 +2,166 @@
 
-    replay_buffer = make_replay_buffer(device, cfg)
-
-    recorder = transformed_env_constructor(
-        cfg,
-        video_tag=video_tag,
-        norm_obs_only=True,
-        obs_norm_state_dict=obs_norm_state_dict,
-        logger=logger,
-        use_env_creator=False,
-    )()
-    if isinstance(create_env_fn, ParallelEnv):
-        raise NotImplementedError("This behaviour is deprecated")
-    elif isinstance(create_env_fn, EnvCreator):
-        recorder.transform[1:].load_state_dict(create_env_fn().transform.state_dict())
-    elif isinstance(create_env_fn, TransformedEnv):
-        recorder.transform = create_env_fn.transform.clone()
-    else:
-        raise NotImplementedError(f"Unsupported env type {type(create_env_fn)}")
-    if logger is not None and video_tag:
-        recorder.insert_transform(0, VideoRecorder(logger=logger, tag=video_tag))
-
-    # reset reward scaling
-    for t in recorder.transform:
-        if isinstance(t, RewardScaling):
-            t.scale.fill_(1.0)
-            t.loc.fill_(0.0)
-
-    trainer = make_trainer(
-        collector,
-        loss_module,
-        recorder,
-        target_net_updater,
-        actor_model_explore,
-        replay_buffer,
-        logger,
-        cfg,
-    )
+    # Create Agent
+    model, exploration_policy = make_ddpg_agent(train_env, eval_env, device)
 
-    final_seed = collector.set_seed(cfg.seed)
-    print(f"init seed: {cfg.seed}, final seed: {final_seed}")
+    # Create Loss Module and Target Updater
+    loss_module, target_net_updater = make_loss_module(cfg, model)
+
+    # Make Off-Policy Collector
+    collector = make_collector(cfg, train_env, exploration_policy)
+
+    # Make Replay Buffer
+    replay_buffer = make_replay_buffer(
+        batch_size=cfg.batch_size,
+        prb=cfg.prb,
+        buffer_size=cfg.buffer_size,
+        device=device,
+    )
 
-    trainer.train()
-    return (logger.log_dir, trainer._log_dict)
+    # Make Optimizers
+    optimizer_actor, optimizer_critic = make_optimizer(cfg, loss_module)
+
+    rewards = []
+    rewards_eval = []
+
+    # Main loop
+    collected_frames = 0
+    pbar = tqdm.tqdm(total=cfg.total_frames)
+    r0 = None
+    q_loss = None
+
+    for i, tensordict in enumerate(collector):
+        # update weights of the inference policy
+        collector.update_policy_weights_()
+
+        if r0 is None:
+            r0 = tensordict["next", "reward"].sum(-1).mean().item()
+        pbar.update(tensordict.numel())
+
+        tensordict = tensordict.view(-1)
+        current_frames = tensordict.numel()
+        replay_buffer.extend(tensordict.cpu())
+        collected_frames += current_frames
+
+        # optimization steps
+        if collected_frames >= cfg.init_random_frames:
+            (
+                actor_losses,
+                q_losses,
+            ) = ([], [])
+            for _ in range(
+                int(cfg.env_per_collector * cfg.frames_per_batch * cfg.utd_ratio)
```

Comment:
we should remove all the `cfg.smth` from the training loop as this can slow down training (it is a known omegaconf issue).

What i usually do:

```python
env_per_collector = cfg. env_per_collector
frames_per_batch = cfg. frames_per_batch
utd_ratio = cfg. utd_ratio

...

    for i, tensordict in enumerate(collector):
            for _ in range(
                int(env_per_collector * frames_per_batch * utd_ratio)
            ):
```